geomstats/stratified_geometry/spider.py

```diff
@@ -0,0 +1,303 @@
+        """Return the hash of the instance."""
+        return gs.array([self.s, self.x])
+
+
+class Spider(PointSet):
+    r"""Spider: a set of rays attached to the origin.
+
+    Parameters
+    ----------
+    rays : int
+        Number of rays to attach to the origin.
+        Note that zero counts as the origin not as a ray.
+    """
+
+    def __init__(self, rays):
+        super(Spider, self).__init__()
+        self.rays = rays
+
+    def random_point(self, n_samples=1):
+        r"""Compute a random point of the spider set.
+
+        Parameters
+        ----------
+        n_samples : int
+            Number of samples.
+            Optional, default: 1.
+
+        Returns
+        -------
+        samples : list of SpiderPoint, shape=[...]
+            List of SpiderPoints randomly sampled from the Spider.
+        """
+        if self.rays != 0:
+            s = gs.random.randint(low=0, high=self.rays, size=n_samples)
+            x = gs.abs(gs.random.normal(loc=10, scale=1, size=n_samples))
+            x[s == 0] = 0
+            return [SpiderPoint(s=s[k], x=x[k]) for k in range(n_samples)]
+        return [SpiderPoint(s=0, x=0)] * n_samples
+
+    @_vectorize_point((1, "point"))
+    def belongs(self, point):
+        r"""Check if a random point belongs to the spider set.
+
+        Parameters
+        ----------
+        point : SpiderPoint or list of SpiderPoint, shape=[...]
+             Point to be checked.
+
+        Returns
+        -------
+        belongs : array-like, shape=[...]
+            Boolean denoting if the SpiderPoint belongs to the Spider Set.
+        """
+        results = []
+        for single_point in point:
+            results += [
+                self._value_check(single_point)
+                and self._rays_check(single_point)
+                and self._zero_check(single_point)
+                and type(single_point) is SpiderPoint
+            ]
+        return gs.array(results)
+
+    def _rays_check(self, single_point):
+        r"""Check if a random point has the correct number of rays.
+
+        Parameters
+        ----------
+        single_point : SpiderPoint
+             Point to be checked.
+
+        Returns
+        -------
+        belongs : boolean
+            Boolean denoting if the point has a ray in the rays set.
+        """
+        if single_point.s not in list(range(self.rays + 1)):
+            return False
+        return True
+
+    @staticmethod
+    def _zero_check(single_point):
+        r"""Check if a random point satisfy the zero condition.
+
+        Parameters
+        ----------
+        single_point : SpiderPoint
+             Point to be checked.
+
+        Returns
+        -------
+        belongs : boolean
+            Boolean denoting if the point has zero length when it has zero ray.
+        """
+        if single_point.s == 0 and single_point.x != 0:
+            return False
+        return True
+
+    @staticmethod
+    def _value_check(single_point):
```

Comment:
This is effectively checking x, i.e. the coordinate on the ray: should it be called _coord_check? 

Using the same name several times is easier for users who are discovering the file.